core/src/main/java/com/linecorp/armeria/common/FlagsProvider.java

```diff
@@ -85,6 +85,15 @@ public interface FlagsProvider {
      */
     int priority();
 
+    /**
+     * Returns whether this {@link FlagsProvider} should be enabled
+     * when calculating {@link Flags}. The default value is {@code true}.
+     */
+    @UnstableApi
+    default boolean enabled() {
```

Comment:
isEnabled?

Reply:
I couldn't immediately understand what it means that flags are disabled.
We already have `priority()` to control and merge the values in different `Flags`.
I think we can leverage that feature to disable timeouts.

`prority()` can be set to `Integer.MAX_VALUE` when `TimeoutDebugUtil.isTimeoutDisabled()` is enabled. Otherwise, `Integer.MIN_VALUE`. The following values need to be adjusted. 
- `DefaultFlagsProvider.priority()`: `Integer.MIN_VALUE + 1`
- `SystemPropertyFlagsProvider.priority()`: `Integer.MAX_VALUE - 1`